edx_exams/apps/core/api.py

```diff
@@ -26,6 +33,22 @@ def get_attempt_by_id(attempt_id):
 
     return None
 
+def get_attempt_by_user_id(user_id):
+    """ 
+    Function to fetch a user's latest exam attempt
+    that has a status of 'started' or 'ready_to_submit'.
+    """
+
+    user_attempts = ExamAttempt.get_attempts_by_user(user_id)
+    
+    in_progress_attempts = user_attempts.filter(status__in=[ExamAttemptStatus.started, ExamAttemptStatus.ready_to_submit])
```

Comment:
I think you should be able to get the most recently created attempt using a django query, as opposed to the if statement you added. Something like https://github.com/edx/edx-exams/blob/0c139a85a90ae34f49e80f05896eb0701473c19e/edx_exams/apps/core/models.py#L176